File: sd/trabalhoFinal/new/despachante.py
import json
from cabecalho import Cabecalho
from esqueleto import Esqueleto

class Despachante:
    def __init__(self):
        self.esqueleto = Esqueleto()

    def invoke(self, request):
        request_data = json.loads(request)
        cabecalho = Cabecalho(**request_data)

        result = None

        if cabecalho.methodId == 1:
            filme = cabecalho.arguments["filme"]
            result = self.esqueleto.adicionarFilme(filme)
        elif cabecalho.methodId == 2:
            id = cabecalho.arguments["id"]
            result = self.esqueleto.removerFilme(id)
        elif cabecalho.methodId == 3:
            id = cabecalho.arguments["id"]
            result = self.esqueleto.exibirDetalhe(id)
        elif cabecalho.methodId == 4:
            result = self.esqueleto.mostrarCatalogo()

        response_data = {
            "result": result
        }
        response = json.dumps(response_data)
        return response

File: sd/trabalhoFinal/final/src/server/Despachante.java
package org.example;

public class Despachante {
    private Database db;

    public Despachante(Database db) {
        this.db = db;
    }

    public Cabecalho invoke(String jsonStr) {
        Cabecalho cabecalho = Cabecalho.fromJson(jsonStr);
        int methodId = cabecalho.getMethodId();
        String arguments = cabecalho.getArguments();
        Esqueleto esqueleto = new Esqueleto(db);
        String responseArguments;

        switch (methodId) {
            case 1:
                responseArguments = esqueleto.adicionarFilme(arguments);
                break;
            case 2:
                responseArguments = esqueleto.removerFilme(arguments);
                break;
            case 3:
                responseArguments = esqueleto.exibirDetalhe(arguments);
                break;
            case 4:
                responseArguments = esqueleto.mostrarCatalogo();
                break;
            default:
                responseArguments = "Método não reconhecido";
                break;
        }

        return new Cabecalho(1, cabecalho.getObjectReference(), methodId, responseArguments);
    }
}